feat: Add Roche Cedex HiRes Adapter

poojabalaji05 opened the pull request:
Add adapter for the Roche Cedex HiRes instrument. The adapter handles both XLSX and CSV exports of the instrument data.
The adapter uses [cell counter](https://github.com/Benchling-Open-Source/allotropy/blob/main/src/allotropy/allotrope/models/adm/cell_counting/benchling/_2023/_11/cell_counting.py) schema of allotropy.

nathan-stender commented on src/allotropy/parsers/roche_cedex_hires/roche_cedex_hires_parser.py:
```diff
@@ -0,0 +1,437 @@
+from allotropy.parsers.vendor_parser import VendorParser
+
+
+def get_property_value(
```
```suggestion
DataType = TypeVar("DataType")

def get_property_value(
    data_frame: pd.DataFrame, column: str, row: int, datatype: DataType
) -> DataType:
```

nathan-stender commented on src/allotropy/parsers/roche_cedex_hires/roche_cedex_hires_parser.py:
```diff
@@ -0,0 +1,437 @@
+                                            )
+                                        ),
+                                        viable_cell_density__cell_counter_=TQuantityValueMillionCellsPerMilliliter(
+                                            value=float(
```
Use `get_calculated_data`?

nathan-stender replied:
You can handle this by wrapping `get_calculated_data` in `assert_not_none` from `allotropy.parsers.utils.values`

poojabalaji05 replied:
Fixed this!

nathan-stender reviewed (changes requested):
Code looks good! 

I couple small requests, otherwise LGTM

nathan-stender commented:
@poojabalaji05 looks like there are some typing issues, with current version, you can view these by running `hatch run lint:typing` or just `hatch run lint`. Will approve once passing!

poojabalaji05 commented:
@nathan-stender 
Fixed the linting issues. Once the [Qubit4 PR](https://github.com/Benchling-Open-Source/allotropy/pull/500) is approved and the code is merged, the test scripts for Roche Cedex HiRes should function correctly upon syncing the forks. This is because we've added the functionality to remove custom fields if their values are `None` in the Qubit4 PR.

nathan-stender commented:
@poojabalaji05 still have a test failure here 👀

nathan-stender commented:
@poojabalaji05 if you pull past this PR: https://github.com/Benchling-Open-Source/allotropy/pull/507, you can run `hatch run scripts:update-readme` to update the README file!

poojabalaji05 commented:
@nathan-stender 
I have updated the `README.md` file by adding the parser and also included the vendor in the `NON_READY_PARSERS` dictionary to avoid test failure in test_vendor_release_state function.
I noticed that the test script structure has been modified, so I have made the necessary adjustments from my end as well.
 
Additionally, the Roche Cedex HiRes test is failing due to `null` values being added to the custom information document dictionary. This issue will be resolved once we sync the fork after merging [Qubit 4 PR](https://github.com/Benchling-Open-Source/allotropy/pull/500), which removes the null keys.

poojabalaji05 commented:
@nathan-stender 
I've pulled the latest changes from the main branch, and all checks are passing. Could you merge this fork for Roche Cedex Hires adapter into the main repository?
